thrift/src/main/java/com/linecorp/armeria/common/thrift/text/StackWalkingThriftMessageClassFinder.java

```diff
@@ -0,0 +1,137 @@
+                                    MethodType.methodType(Object.class, Function.class));
+
+        final Class<?> stackFrameClass = classLoader.loadClass("java.lang.StackWalker$StackFrame");
+        final String invokingFailMsg = "Failed to invoke StackWalker.StackFrame.getDeclaringClass():";
+        Function<Object, Class<?>> getClassByStackFrameTemp;
+        Object instance;
+
+        try {
+            // StackWlaker instantiate with RETAIN_CLASS_REFERENCE option
+            final Class<?> Option = classLoader.loadClass("java.lang.StackWalker$Option");
+            final MethodHandle getInstanceMH =
+                    lookup.findStatic(stackWalkerClass,
+                                      "getInstance",
+                                      MethodType.methodType(stackWalkerClass, Option));
+            final Enum<?> RETAIN_CLASS_REFERENCE = Arrays.stream((Enum<?>[]) Option.getEnumConstants())
+                    .filter(op -> "RETAIN_CLASS_REFERENCE".equals(op.name())).findFirst().orElseGet(null);
+
+            if (RETAIN_CLASS_REFERENCE == null) {
+                throw new IllegalStateException("Failed to get RETAIN_CLASS_REFERENCE option");
+            }
+            instance = getInstanceMH.invoke(RETAIN_CLASS_REFERENCE);
+            final MethodHandle getDeclaringClassMH = lookup.findVirtual(stackFrameClass,
+                                                                "getDeclaringClass",
+                                                                MethodType.methodType(Class.class));
+
+            getClassByStackFrameTemp = stackFrame -> {
+                try {
+                    return getMatchedClass((Class<?>) getDeclaringClassMH.invoke(stackFrame));
+                } catch (Throwable t) {
+                    logger.warn(invokingFailMsg, t);
+                }
+                return null;
+            };
+        } catch (Throwable throwable) {
+            // StackWlaker instantiate without option
+            logger.warn("Falling back to StackWalker without option:", throwable);
+            final MethodHandle getInstanceMH =
+                    lookup.findStatic(stackWalkerClass,
+                                      "getInstance",
+                                      MethodType.methodType(stackWalkerClass));
+            final MethodHandle getClassNameMH =
+                    lookup.findVirtual(stackFrameClass, "getClassName", MethodType.methodType(String.class));
+
+            instance = getInstanceMH.invoke();
+            getClassByStackFrameTemp = stackFrame -> {
+                try {
+                    return getMatchedClass(getClassByName(getClassNameMH.invoke(stackFrame).toString()));
+                } catch (Throwable t) {
+                    logger.warn(invokingFailMsg, t);
+                }
+                return null;
+            };
+        }
+
+        stackWalker = instance;
+
+        final Function<Object, Class<?>> getClassByStackFrame = getClassByStackFrameTemp;
```

Comment:
Can we make `getClassByStackFrameTemp` final? Then we would not need this local variable.